cansat-stm32f446/src/blink.rs

```diff
@@ -0,0 +1,28 @@
+//! Blinking LED using  blocking `Delay` in  TIM5.
+#![deny(unsafe_code)]
+
+use crate::hal::{pac, prelude::*};
+use panic_halt as _; // panic handler
+
+pub fn blink_loop() {
+    if let (Some(dp), Some(cp)) = (
+        pac::Peripherals::take(),
+        cortex_m::peripheral::Peripherals::take(),
+    ) {
```

Comment:
when one of the `take` functions fails, the entire if-expression is skipped and we basically halt the system. I'd be less noisy to this:
```suggestion
    let dp = pac::Peripherals::take().unwrap();
    let cp = cortex_m::peripheral::Peripherals::take().unwrap();
```
Program running correctly should be our "default" program path. So, the following:
```rust
if is_error {
    handle_error();
}
do_useful_stuff();
```
is much more natural than:
```rust
if !is_error {
    do_useful_stuff();
}
handle_error();